### ISSUE TITLE
Animated Title: "Dissolving Text" is not working properly when Animation Length != default

### ISSUE THREAD

**Issue by bobonov** [OpenShot/openshot-qt#3356]
**Describe the bug:**
When creating a Dissolving Text animated title and setting animation length to a value different from default it results in an incorrect animation.
The result seems that have the animation frames mixed, I attach the resulting video (all other parameter are left as default)
[export.zip](https://github.com/OpenShot/openshot-qt/files/4444094/export.zip)

**Side bug:**
Changing the animation length do not update the total frames
![UmcYRLnM7Y](https://user-images.githubusercontent.com/1412083/78668079-5acfe880-78da-11ea-94c8-0fad168462e6.png)

**Steps to reproduce the behavior:**

1. Go to Animated Title
2. Select "dissolving Text" effect
3. Select a value different from default for Animation Length (the attached video is with 2x)
4. Click on render

**Expected behavior:**
A proper animated title

**System Details:**
- OpenShot-v2.5.1-dev2-1586151841-166f6938-232f3f79-x86_64
- Operating System / Distro: Windows 10
- Blender: 2.82

**Comment by SuslikV** [comment]
Use _notepad++_ application (minimalist version is enough for your needs) open in it next `.py` file (from your OpenShot program folder):
`C:\Program Files\OpenShot Video Editor\windows\views\blender_listview.py`. Find in the file `blender_listview.py` next block of code:
```
    def dropdown_index_changed(self, widget, param, index):
        value = widget.itemData(index)
        self.params[param["name"]] = value
        log.info('Animation param %s set to %s' % (param["name"], value))
```
add `self.init_slider_values()` below, like:
```
    def dropdown_index_changed(self, widget, param, index):
        value = widget.itemData(index)
        self.params[param["name"]] = value
        log.info('Animation param %s set to %s' % (param["name"], value))
        self.init_slider_values()
```
Python code is sensitive to spaces! And they are differs from tabs!

Save changes and start OpenShot. Is it better now?

**Comment by bobonov** [comment]
> add self.init_slider_values()
This solves only the slider problem

What about the main bug?

**Comment by ferdnyc** [comment]
@SuslikV That's an incomplete fix, I'm afraid, as it won't fix the render progress display.

I'll have a PR with a complete fix shortly, I'm also correcting another bug I noticed where the animation is alway s being imported at 25fps, instead of (like it should be) the project frame rate.

**Comment by bobonov** [comment]
Thanks in advance
I'm looking at the dir .openshot_qt\blender\OL9Z8CPGNK which has the png files name for the animation sequence looks mixed I think it start after the frame 128.

**Comment by SuslikV** [comment]
> What about the main bug?

What the main bug is?

**Comment by ferdnyc** [comment]
Hmm. Well, my attempt at fixing the frame rate didn't work, but that's secondary. The animation-duration length fix is still there.

@SuslikV Basically, this:
![ezgif com-optimize](https://user-images.githubusercontent.com/538020/78692223-901f0b00-78c7-11ea-9360-0083e12c2499.gif)

The frames are getting mixed together. I _believe_ my duration fix will solve that as well, though I have to test to make sure.

I'll also have an installer built from the PR branch shortly, that I can share for testing.

**Comment by ferdnyc** [comment]
Wow, that is annoying, and I'm vaguely worried about causing problems for anyone strobe-sensitive who views this. Maybe I'll hide it behind a details tag.

**Comment by ferdnyc** [comment]
The bug with dissolve is, I think, actually the same bug #3353 addressed — the particle lifetime was too short, so they were resetting to their initial places before the animation completed.

The change I've made in #3357 _might_ not fix it. (I'm still waiting on a render to find out.) The animation-length feature uses Blender's frame remapping to extend the length of the animation, so the question is whether the particle lifetimes need to take that into account. If the lifetimes need to be extended based on the frame mapping, that'll take some more work to fix.

I won't know for an hour or so, my computer is very slow and rendering 256 frames of particle physics takes for-fucking-ever. But I'll update when I find out.

The problem there was that the

**Comment by ferdnyc** [comment]
> I think it start after the frame 128.

Hrm. I'd feel better if you'd said 180. Frame 90 was the point where particles previously started resetting, and bug #3350 was triggered (the one I fixed with #3353).

But if it's starting after frame 128 (the end of the un-remapped animation), then the remapping feature might be broken in general when particle physics are employed, and we may have to find another way of extending the length of rendered animations.

Although... #3353 extended particle lifetimes from 80 frames to 120 (after they're generated beginning in frame 10) — so, if the problem starts around frame 130, it could still just be a lifetime issue.

Anyway, my title-generation job is on frame 99, so we'll know someday.

**Comment by ferdnyc** [comment]
Oh, yeah, here's a link to the Win64 installer of a Daily Build with #3357 incorporated, in case anyone wants to test it in its current state. (If I end up needing to make further changes, I'll have to provide a new download.)

https://drive.google.com/file/d/1mJFFwxUfg8fD0tolkjvqttM7T-5uT5Aw/view?usp=sharing

**Comment by ferdnyc** [comment]
Well, this is interesting.

The issue is still present, and appears to be _much_ bigger than a simple lifetime issue — using frame remapping to extend the animation length appears not to work correctly, at least when particle systems are involved.

You can see here that, by frame 63/256, there are already effectively two animations running at once: One on the odd frames, and a completely different one on the even frames. It's subtle, at _this_ point, but only grows worse as the animation progresses. 

![image](https://user-images.githubusercontent.com/538020/78707137-34ac4780-78de-11ea-87cb-17db2cf974b2.png)

I'll see if I can work out a fix, but we may just have to disable the "Animation Length" option for the time being, in the title templates that use the particle physics engine.

**Comment by ferdnyc** [comment]
OK, I believe I've solved this. It'll require slightly more extensive changes to the render setup than I'd originally planned.

For any renders using particle physics, the animation parameters will have to be _generated_ for the full length of the animation, rather than setting them up for only the default length and using frame remapping.

As a result, previews for frame N also can't be generated by setting the start and end frame to N (since the "true" length of the animation in start and end frames is needed as a parameter, even to generate previews), so I also have to update _all_ of the Blender scripts with that change. But I'm glad to be making that change, as the preview thing was always a hack.

I also, in doing this, came up with a _better_ fix for #3350, since as it turns out the tendency of particles to "reset" to their starting position after the end of their lifetime is optional, and just turning that off is better than increasing the lifetime. With `show_dead = False` and the original shorter lifetime, particles won't just hang around on screen even right up to the end of the animation. Instead, they'll die off gradually over time, as intended.

I'll update the PR once I have all of these changes in place.

**Comment by ferdnyc** [comment]
The PR has been updated. I ended up making significant further changes to the entire animated title setup/rendering system, so here's an updated Win64 installer with those changes included:

https://drive.google.com/file/d/1W0-2O0Vn4fn4OucdhVNmR_2LAn5kGiiU/view?usp=sharing

**Comment by ferdnyc** [comment]
> For any renders using particle physics, the animation parameters will have to be _generated_ for the full length of the animation, rather than setting them up for only the default length and using frame remapping.

I wound up backing off on that, in favor of just regenerating the animation parameters for the remapped frames. I was hoping to build the physics-based renders to be a bit more intelligent, so that (for example) I could have `explode.py` set up the animation parameters so that the explosion _always_ started on frame 48, and the animation length parameter merely extended the duration. But that proved too complicated, so I went back to the old setup based on frame remapping, which means that for longer animation lengths, all events will be proportionally spread out (just like they have been in the past).

I did rework how previews are generated, though, in a way that makes a lot more sense. It also avoids the "baking" process for physics animations when generating just a preview frame, which will make previewing a lot faster for physics animations with an "Animation Length" parameter set.
